assets/src/adaptivity/rules-engine.ts

```diff
@@ -306,31 +306,34 @@ export const check = async (
   }
 
   let score = 0;
-  if (scoringContext.trapStateScoreScheme) {
-    // apply all the actions from the resultEvents that mutate the state
-    // then check the session.currentQuestionScore and clamp it against the maxScore
-    // setting that value to score
-    const mutations = resultEvents.reduce((acc, evt) => {
-      const { actions } = evt.params as Record<string, any>;
-      const mActions = actions.filter(
-        (action: any) =>
-          action.type === 'mutateState' && action.params.target === 'session.currentQuestionScore',
-      );
-      return acc.concat(...acc, mActions);
-    }, []);
-    if (mutations.length) {
-      const mutApplies = mutations.map(({ params }) => params);
-      bulkApplyState(mutApplies, env);
-      score = getValue('session.currentQuestionScore', env) || 0;
+  if (isCorrect || scoringContext.negativeScoreAllowed) {
```

Comment:
im not sure that isCorrect should be involved here? or if negativeScoreAllowed? I'm non following the business rules logic, maybe we can explain it in a comment or something

Reply:
I didnt acually got your point.. however i have added comments. let me know if that helps.

Reply:
well I think one thing that I'm nervous about is that I think this will prevent mutations from updating the current score how they see fit unless it is correct. so it could be an incorrect with a mutation to change current score and that wouldn't work. also, the other scoring method that is based on attempts won't be able to save the in progress score... maybe that's what the issue is? hmm yes, looking at the ticket they just don't want to *see* the score until it is correct... hmmm OK well I guess then I can just assume that they won't be using mutate unless negativeScore is allowed...